core/lib/runner.js

```diff
@@ -503,21 +512,21 @@ function $randomString(length) {
   return Math.random().toString(36).substr(2, length);
 }
 
-function handleBeforeRequests(script, runnableScript, runnerEngines, testEvents) {
+function handleScriptHook(hook, script, engines, hookEvents, contextVars) {
   let ee = new EventEmitter();
   return new Promise(function(resolve, reject){
     ee.on('request', function() {
-      testEvents.emit('beforeTestRequest');
+      hookEvents.emit(`${hook}ScriptHookRequest`);
     });
     ee.on('error', function(error) {
-      testEvents.emit('beforeTestError', error);
+      hookEvents.emit(`${hook}ScriptHookError`, error);
```

Comment:
I am not sure if we should change the names of these events as existing custom engines may rely on them

Reply:
True. I just reverted those.